kebechet/managers/thoth_provenance/thoth_provenance.py

```
@@ -0,0 +1,101 @@
+from kebechet.exception import InternalError
+from kebechet.exception import PipenvError
+from kebechet.managers.manager import ManagerBase
+from kebechet.source_management import Issue
+from kebechet.source_management import MergeRequest
+from kebechet.utils import cloned_repo
+
+
+_LOGGER = logging.getLogger(__name__)
+
+_BRANCH_NAME = "kebechet_thoth"
+
+
+class ThothProvenanceManager(ManagerBase):
+    """Manage source issues of dependencies."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize ThothProvenance manager."""
+        self._cached_merge_requests = None
+        super().__init__(*args, **kwargs)
+
+    @property
+    def sha(self):
+        """Get SHA of the current head commit."""
+        return self.repo.head.commit.hexsha
+
+    def _issue_provenance_error(self, prov_results: list, labels: list):
+        error_info = prov_results[0]
+        text_block = ""
+
+        for err in error_info:
+            _LOGGER.info(f"{err['id']}: {err['package_name']} {err['package_version']}")
+            text_block = (
+                text_block
+                + f"## {err['type']}: {err['id']} - {err['package_name']}:{err['package_version']}\n"
+                + f"**Justification: {err['justification']}**\n"
+                + f"#### source: \n {pprint.pformat(err['source'])}\n"
+                + f"#### lock_info:\n {pprint.pformat(err['package_locked'])}"
+            )
+
+        checksum = hashlib.md5(text_block.encode("utf-8")).hexdigest()[:10]
+
+        self.sm.open_issue_if_not_exist(
+            f"{checksum} - {len(error_info)}: Automated kebechet thoth-provenance Issue",
+            lambda: text_block,
+            labels=labels,
+        )
+
+    def run(self, labels: list):
+        """Run the provenance check bot."""
+        with cloned_repo(self.service_url, self.slug, depth=1) as repo:
+            self.repo = repo
+            if os.path.isfile("Pipfile") and os.path.isfile("Pipfile.lock"):
```

If not? I would expect an issue created on my repository, that bot cannot operate on it instead of silently skipping checks (which can be dangerous sometimes).